internal/backoff/backoff.go

```diff
@@ -60,12 +62,18 @@ func New(opts ...Option) Backoff {
 	return b
 }
 
-func (b *backoff) Do(ctx context.Context, f func() (val interface{}, retryable bool, err error)) (res interface{}, err error) {
-	res, ret, err := f()
+func (b *backoff) Do(ctx context.Context, f func(ctx context.Context) (val interface{}, retryable bool, err error)) (res interface{}, err error) {
+	res, ret, err := f(ctx)
 	if err == nil || !ret {
 		return
 	}
 
+	ctx, span := trace.StartSpan(ctx, "vald/internal/backoff/Backoff.Do")
+	defer func() {
+		if span != nil {
+			span.End()
+		}
+	}()
```

Comment:
I think this block should be placed before L66.

Reply:
Exactly, I'm considered about it, but first execution is not backoff, do we need to trace first execution?

Reply:
in the current impl, execution time of retries can be traced by `vald/internal/backoff/Backoff.Do/${cnt}` spans but the execution time of the first one cannot be traced. if we include the first one in the `vald/internal/backoff/Backoff.Do` span, we can calculate it from the traces.

if we adopt the current impl, the name of the span should be renamed as `vald/internal/backoff/Backoff.Do/backoff-executions` because the name `vald/internal/backoff/Backoff.Do` leads misunderstanding that it covers the function `Do` entirely.

Reply:
If we trace a function, we should implement the trace code inside the function, so we don't need to trace the execution time of the first function, and it is more efficient to trace only the case of retry execution so that the nesting of the trace does not become deep, so I'm going to rename trace tags but `vald/internal/backoff/Backoff.Do/backoff-executions` sounds redundant how about `vald/internal/backoff/Backoff.Do/retry` or `vald/internal/backoff/Backoff.Do/execute`

Reply:
okay.
`vald/internal/backoff/Backoff.Do/retry` and `vald/internal/backoff/Backoff.Do/retry/${i}` sounds better.

Reply:
ok! I'll fix ti